mayastor/src/grpc/mayastor/v1.rs

```diff
@@ -0,0 +1,196 @@
+use crate::{
+    core::Share,
+    grpc::{rpc_submit, GrpcClientContext, GrpcResult, Serializer},
+    lvs::{Error as LvsError, Lvs, PoolArgs},
+    subsys::PoolConfig,
+};
+use futures::FutureExt;
+use nix::errno::Errno;
+use std::{convert::TryFrom, fmt::Debug, time::Duration};
+use tonic::{Request, Response, Status};
+
+use rpc::mayastorv1::*;
+
+#[derive(Debug)]
+struct UnixStream(tokio::net::UnixStream);
+
+use ::function_name::named;
+use std::panic::AssertUnwindSafe;
+
+#[derive(Debug)]
+pub struct MayastorSvc {
+    name: String,
+    interval: Duration,
+    rw_lock: tokio::sync::RwLock<Option<GrpcClientContext>>,
```

Comment:
The content of the lock are more important than the lock type itself so I'd name identifiers after the data, so `client_context` would be better IMO.

Reply:
Also, why a `RwLock` if you never take a read lock on it? :thinking: